### PR TITLE
Add Prewitt and Sobel filter

### PR DESCRIPTION
**Pull request by yongtang**
This PR adds Prewitt and Sobel filter under `tfio.experimental.filter[prewitt|sobel]`

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

### CODE REVIEW

File: tests/test_filter_eager.py [hunk 1]
```diff
@@ -63,8 +63,46 @@
...
             ),
         ),
+        pytest.param(
+            lambda: (np.reshape(np.arange(25) / 25, [1, 5, 5, 1])),
+            lambda e: tfio.experimental.filter.prewitt(e),
+            lambda e: np.reshape(
+                np.sqrt(
+                    np.square(
+                        scipy.ndimage.prewitt(
+                            np.reshape(e, [5, 5]), axis=0, mode="constant"
```
Comment (terrytangyuan) [inline tests/test_filter_eager.py:73]: Do we want to test multiple modes here?

**Comment by yongtang** [comment]
@terrytangyuan Different modes can be grouped into a separate tests with all ops suitable for padding. Maybe we can leave this PR alone? I will create a follow up PR for modes.

**Review by terrytangyuan — APPROVED** [review]
Sounds good.

File: tensorflow_io/core/python/experimental/filter_ops.py [hunk 2]
```diff
@@ -243,3 +243,83 @@ def gabor(
...
+    """
+    Apply Prewitt filter to image.
+
+    Args:
+      input: A 4-D (`[N, H, W, C]`) Tensor.
+      mode: A `string`. One of "CONSTANT", "REFLECT", or "SYMMETRIC"
+        (case-insensitive). Default "CONSTANT".
+      constant_values: A `scalar`, the pad value to use in "CONSTANT"
+        padding mode. Must be same type as input. Default 0.
+      name: A name for the operation (optional).
```
Comment (kvignesh1420) [inline tensorflow_io/core/python/experimental/filter_ops.py:258]: @yongtang just wanted to know how we will be using the `name` parameter in the future. (like naming the OP etc.)

**Review by kvignesh1420 — APPROVED** [review]

**Comment by yongtang** [comment]
@kvignesh1420 The name is normally used with `tf.name_scope` See https://www.tensorflow.org/api_docs/python/tf/name_scope. It can also be passed to C++ kernel's `name` attributes (All C++ kernels by default generates a `name` attr once compiled). In tensorflow-io we haven't systematically pass `name` as needed. We should as we gradually improve the code base.

**Comment by kvignesh1420** [comment]
@yongtang thanks for the info.

**Pull request merged**